test/functional/wallet_tokens_change_supply.py

```diff
@@ -0,0 +1,193 @@
+        node = self.nodes[0]
+
+        # new wallet
+        async with WalletCliController(node, self.config, self.log) as wallet:
+            await wallet.create_wallet()
+
+            # check it is on genesis
+            assert_equal('0', await wallet.get_best_block_height())
+
+            # new address
+            pub_key_bytes = await wallet.new_public_key()
+            assert_equal(len(pub_key_bytes), 33)
+
+            # Get chain tip
+            tip_id = node.chainstate_best_block_id()
+            self.log.debug(f'Tip: {tip_id}')
+
+            # Submit a valid transaction
+            output = {
+                    'Transfer': [ { 'Coin': 1001 * ATOMS_PER_COIN }, { 'PublicKey': {'key': {'Secp256k1Schnorr' : {'pubkey_data': pub_key_bytes}}} } ],
+            }
+            encoded_tx, tx_id = make_tx([reward_input(tip_id)], [output], 0)
+
+            node.mempool_submit_transaction(encoded_tx)
+            assert node.mempool_contains_tx(tx_id)
+
+            block_id = self.generate_block() # Block 1
+            assert not node.mempool_contains_tx(tx_id)
+
+            # sync the wallet
+            assert_in("Success", await wallet.sync())
+
+            # check wallet best block if it is synced
+            assert_equal(await wallet.get_best_block_height(), '1')
+            assert_equal(await wallet.get_best_block(), block_id)
+
+            assert_in("Coins amount: 1001", await wallet.get_balance())
+
+            address = await wallet.new_address()
+
+            # invalid ticker
+            # > max len
+            token_id, err = await wallet.issue_new_token("asdddd", "10000", 2, "http://uri", address)
+            assert token_id is None
+            assert err is not None
+            assert_in("Invalid ticker length", err)
+            # non alphanumeric
+            token_id, err = await wallet.issue_new_token("asd#", "10000", 2, "http://uri", address)
+            assert token_id is None
+            assert err is not None
+            assert_in("Invalid character in token ticker", err)
+
+            # invalid url
+            token_id, err = await wallet.issue_new_token("XXX", "10000", 2, "123 123", address)
+            assert token_id is None
+            assert err is not None
+            assert_in("Incorrect metadata URI", err)
+
+            # invalid num decimals
+            token_id, err = await wallet.issue_new_token("XXX", "10000", 99, "http://uri", address)
+            assert token_id is None
+            assert err is not None
+            assert_in("Too many decimals", err)
+
+            # issue a valid token
+            token_id, err = await wallet.issue_new_token("XXX", "10000", 2, "http://uri", address, 'lockable')
+            assert token_id is not None
+            assert err is None
+            self.log.info(f"new token id: {token_id}")
+
+            self.generate_block()
+            assert_in("Success", await wallet.sync())
+
+            tokens_to_mint = 10000
```

Comment:
Can we randomize this test? I'm OK with making another copy that would do the randomization.

Reply:
specified the --randomseed for the functional tests, and randomized 10 mints/unimts with random amounts